libvast/src/system/index.cpp

```diff
@@ -49,17 +56,683 @@
 
 #include <chrono>
 #include <deque>
+#include <memory>
 #include <unordered_set>
 
+#include "caf/error.hpp"
+#include "caf/response_promise.hpp"
+
 using namespace std::chrono;
 
+// clang-format off
+//
+// The index is implemented as a stream stage that hooks into the table slice
+// stream coming from the importer, and forwards them to the current active
+// partition
+//
+//              table slice              table slice                      table slice column
+//   importer ----------------> index ---------------> active partition ------------------------> indexer
+//                                                                      ------------------------> indexer
+//                                                                                ...
+//
+// At the same time, the index is also involved in the lookup path, where it
+// receives an expression and loads the partitions that might contain relevant
+// results into memory.
+//
+//      expression                atom::evaluate
+//     ------------>  index     ----------------->   partition
+//
+//                                   [indexer]
+//                  (spawns     <----------------     
+//                   evaluator) 
+//
+//                                  curried_predicate
+//                   evaluator  -------------------------------> indexer
+//
+//                                            ids
+//                              <------------------------------
+// clang-format on
+
 namespace vast::system {
 
+namespace v2 {
```

Comment:
I'll append it as a separate commit, so we get one consisting *only* of the removal, otherwise i fear the diff will beecome quite hard to read as the diff algorithm searches for similarities between the v1 and v2 index.